### ISSUE TITLE
[HOLD group chats] Harmonize "name" updates for Rooms and Group Chats. Deprecate non-system message update.

### ISSUE THREAD

**Issue by marcaaron** [Expensify/App#40732]
Coming from https://expensify.slack.com/archives/C05RECHFBEW/p1713436980732419

cc @shawnborton 

## Problem

Workspace rooms and Group chats both provide participants with  ability to give a custom name and update it later. For the Group Chats case, when a name changes there is also no indication of who changed it.

Workspace rooms have this feature, but they use a different style of message that we are used to using (a.k.a. the system message format). 

![Image](https://github.com/Expensify/App/assets/32969087/d5d3dee7-a9b8-48b0-92ab-d4e03c4db430)

We'd really like this to look more like this:

![Image](https://github.com/Expensify/App/assets/32969087/11e4a3c9-9c9f-4048-8278-e3764f524ae0)

## Solution

Are there other system messages that have inconsistent style? Let's fix those too so that they all indicate who took the action and everyone can see who did it.

This is a new feature so needs an engineer to pick up as a small project:

- Make necessary backend changes so that the system message is created.
- Find a contributor to deprecate the old style system message once we have deprecated them completely.

**Comment by dannymcclain** [comment]
Hmm yeah that's a good point. I guess I don't feel super strongly about _not_ including it. I guess I was just thinking that it may be kinda long—but I don't know if that should really be a concern. I would be totally down to include it like we would do with the room/group name.

**Comment by dubielzyk-expensify** [comment]
Yeah I don't actually mind including what the description was set to. Allows users to not having to navigate to see it too. But don't feel too strongly.

**Comment by dannymcclain** [comment]
Ok cool. In that case I would say if we incorporate that update into this, let's just show what the description was changed to!

**Comment by JmillsExpensify** [comment]
Agree with showing when a description was changed.

**Comment by Christinadobrzyn** [comment]
Just checking on this as we're missing an external/internal label. I think this can be external @marcaaron can you confirm?

**Comment by JmillsExpensify** [comment]
@Christinadobrzyn system messages are backend. I hope you don't mind, but I'm going to take this one over as we try to finish up the initial scope of the design doc.

**Comment by dannymcclain** [comment]
Any update on this one?

**Comment by dannymcclain** [comment]
@JmillsExpensify anything new for this?

**Comment by JmillsExpensify** [comment]
Ah, sorry in what sense. I think I might have misunderstand this comment from you. 

 - 

> @Expensify/design should we try to roll image and description updates into this as well? I'm thinking for those two, the system message could just be:
> 
> > **Username**
> > changed the description for this group.
> 
> > **Username**
> > changed the avatar of this group.
> 
> Or something like that. Basically I'm just thinking something slightly more generic/that doesn't include what the value was actually changed to. (I don't feel _too_ strongly, just throwing it out there!)

**Comment by dannymcclain** [comment]
We landed on showing the description.

So I think we'd have something like this:

>**Username**
>changed the name of this group to "Grand Canyon Crew".

>**Username**
>set the description of this group to "We're going to the Grand Canyon! Let's chat about all our plans and split all our expenses here in this group. It's going to be fun on a bun!"

>**Username**
>changed the avatar for this group.

Basically just following the same patterns we use for expenses where every change/update triggers a system message that creates a "history trail" of what's happened.

**Comment by JmillsExpensify** [comment]
Gotcha. For this one...

>Username
> renamed this group from "Danny, Jason, Shawn" to "Grand Canyon Crew".

Would this be more consistent?

>Username
> renamed this group to "Grand Canyon Crew" (previously "Danny, Jason, Shawn")

**Comment by dannymcclain** [comment]
Yeah that works! I was just going off of what the current message says:
![CleanShot 2024-06-05 at 13 11 41@2x](https://github.com/Expensify/App/assets/7516624/965475e6-76a2-4d4f-92fd-723b5ed55853)

But that doesn't really matter. I agree what you've proposed is more consistent with how we handle other updates.

**Comment by JmillsExpensify** [comment]
Cool, sounds good. So then I think we're good to get this one kicked off when an Expensify engineer is ready to pick it up.

**Comment by JmillsExpensify** [comment]
Same as above.

**Comment by dannymcclain** [comment]
Same same Melv.

**Comment by dannymcclain** [comment]
Same as above—still waiting for the right time.

**Comment by dannymcclain** [comment]
Same.

**Comment by dannymcclain** [comment]
I feel like I saw a PR or something recently for updating the room/group name changes? Either way, this is not a super high priority right now.

**Comment by dannymcclain** [comment]
@JmillsExpensify any material update on this? Is this kinda being worked on in other places or are we still just kinda waiting with it?

**Comment by JmillsExpensify** [comment]
I actually think this is fixed in a separate issue now?

**Comment by dannymcclain** [comment]
Looks like for rooms it is fixed.
![CleanShot 2024-09-02 at 09 55 57@2x](https://github.com/user-attachments/assets/afc472e3-c848-40d9-a7b6-32e12355b5b1)

But when I renamed a group, I don't see any system message or note of the change at all.

**Comment by dannymcclain** [comment]
Not overdue, basically still just waiting for the right time.

**Comment by JmillsExpensify** [comment]
Still held on group chats implementation.

**Comment by JmillsExpensify** [comment]
Same

**Comment by dannymcclain** [comment]
Still on hold Melv.

**Comment by dannymcclain** [comment]
![hold](https://github.com/user-attachments/assets/1f02b6e7-7a04-4145-871c-288edf1255f0)

**Comment by dannymcclain** [comment]
Still holding Melv.

**Comment by JmillsExpensify** [comment]
Same same

**Comment by dannymcclain** [comment]
Same³

**Comment by JmillsExpensify** [comment]
All the same

**Comment by dannymcclain** [comment]
... At _some_ point, we should really do this 😂

**Comment by JmillsExpensify** [comment]
I agree, though I also don't think it'll be for a while. How about I add a todo in the [tracking issue](https://github.com/Expensify/Expensify/issues/317715) and we close this issue for now. (Go ahead and close if you agree).
